pkg/plugins/resources/gitea/branch/source.go

```diff
@@ -0,0 +1,47 @@
+package branch
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/result"
+	"github.com/updatecli/updatecli/pkg/plugins/utils/version"
+)
+
+func (g *Gitea) Source(workingDir string) (string, error) {
+	versions, err := g.SearchBranches()
+
+	if err != nil {
+		return "", err
+	}
+
+	if len(versions) == 0 {
+		logrus.Infof("%s No Gitea branches found", result.ATTENTION)
+		return "", nil
+	}
+
+	g.foundVersion, err = g.Spec.VersionFilter.Search(versions)
+
+	if err != nil {
+		switch err {
+		case version.ErrNoVersionFound:
+			logrus.Infof("%s No Gitea branches found matching pattern %q", result.FAILURE, g.versionFilter.Pattern)
+			return "", nil
```

Comment:
I am now wondering if I should return an error or not :/ 
If I don't return an error then a pipeline output looks like 

```
#####################
# TEST GITEA BRANCH #
#####################


SOURCES
=======

default
-------
Searching for version matching pattern "~0.99"
✗ No Gitea branches found matching pattern "~0.99"

⚠ empty source returned


CONDITIONS:
===========

default
-------
✔ Gitea branch "main" found

=============================

REPORTS:


✔ Test Gitea branch:
	Source:
		⚠ [default]  (kind: gitea/branch)
	Condition:
		✔ [default]  (kind: gitea/branch)

Run Summary
===========
Pipeline(s) run:
  * Changed:	0
  * Failed:	0
  * Skipped:	0
  * Succeeded:	1
  * Total:	1

```

Reply:
If I return an error then the pipeline stop right after the source

Reply:
In the case of the github release, it returns an error

Reply:
After consideration, I went to the return error like githubreleas

Reply:
I vote for the error as well. I expect my updatecli pipelines to fail if an empty source is returned.